File: task_equivalents/llm_translations/gpt/optimized/python/statistical_modelling_analysis/02_logistic_regression_odds_ratios.py
import pandas as pd
import statsmodels.api as sm

weekly = pd.read_csv("data/csv/Weekly.csv", index_col=0)

x01 = weekly.iloc[:, 2:8].copy()
y01 = (weekly["Direction"] == "Up").astype(int)

x01 = sm.add_constant(x01)
glm0_fit = sm.GLM(y01, x01, family=sm.families.Binomial()).fit()
print(glm0_fit.summary())

prob = glm0_fit.predict(x01)
pred = (prob > 0.5).astype(int)
print(pd.crosstab(y01, pred, rownames=["Actual"], colnames=["Predicted"]))
print((pred == y01).mean())

File: task_equivalents/llm_translations/gpt/optimized/r/statistical_modelling_analysis/02_logistic_regression_odds_ratios.R
library(readr)

Weekly <- read_csv("data/csv/Weekly.csv", show_col_types = FALSE)

x01 <- Weekly[, 3:8, drop = FALSE]
y01 <- ifelse(Weekly$Direction == "Up", 1, 0)

glm0_fit <- glm(y01 ~ ., data = x01, family = binomial())
print(summary(glm0_fit))

prob <- predict(glm0_fit, type = "response")
pred <- ifelse(prob > 0.5, 1, 0)

print(table(y01, pred))
print(mean(pred == y01))